thirdparty/onnx/CMakeLists.txt

```diff
@@ -18,10 +18,14 @@ else()
     set(ONNX_USE_LITE_PROTO_DEFAULT ON)
 endif()
 
+# from onnx==1.13 it requires C++17 when compiling on Windows
+if(WIN32)
```

Comment:
why not under MSVC?

Reply:
It's just a test for now - it was under MSVC in the previous commit, but ONNX was still being compiled with C++14. I switched to this `if` to see if it works now.

Reply:
I moved the check back under MSVC, it compiles now. The PR is still blocked by a bug with ONNX serialization related to protobuf upgrade.